windows_containers/windows-containers-release-notes-10-16-x.adoc

```diff
@@ -0,0 +1,101 @@
+
+Previously, the WMCO performed a sanitization step that incorrectly replaced commas with semicolons in the user's cluster-wide proxy configuration. This caused Windows to ignore the values set in the `noProxy` environment variable. As a consequence, the WMCO incorrectly sent traffic through the proxy for the endpoints specified in the `no-proxy` parameter. With this fix, the sanitization step that replaced commas with semicolons was removed. As a result, web requests from a Windows node to a cluster-internal endpoint or an endpoint that exists in the `no-proxy` parameter does not go through the proxy. (link:https://issues.redhat.com/browse/OCPBUGS-24264[*OCPBUGS-24264*])
+
+// Copied from 4.15
+Previously, because of bad logic in the networking configuration script, the WMCO was incorrectly reading carriage returns in the containderd CNI configuration file as changes, and identified the file as modified. This caused the CNI configuration to be unnecessarily reloaded, potentially resulting in container restarts and brief network outages. With this fix, the WMCO now reloads the CNI configuration only when the CNI configuration is actually modified. (link:https://issues.redhat.com/browse/OCPBUGS-2887[*OCPBUGS-2887*])
+
+// Copied from 4.12
+Previously, because of routing issues present in Windows Server 2019, under certain conditions and after more than one hour of running time, workloads on Windows Server 2019 could have experienced packet loss when communicating with other containers in the cluster. This fix enables Direct Server Return (DSR) routing within kube-proxy. As a result, DSR now causes request and response traffic to use a different network path, circumventing the bug within Windows Server 2019. (link:https://issues.redhat.com/browse/OCPBUGS-26761[*OCPBUGS-26761*])
+
+// Copied from 4.15
+Previously, the kubelet on Windows nodes was unable to authenticate with private Elastic Container Registries (ECR) registries. Because of this error, the kubelet was not able to pull images from these registries. With this fix, the kubelet is able to pull images from these registries as expected. (link:https://issues.redhat.com/browse/OCPBUGS-26602[*OCPBUGS-26602*])
+
+// Copied from 4.15
+Previously, on Azure clusters the WMCO would check if an external Cloud Controller Manager (CCM) was being used on the cluster. If a CCM is being used, the Operator would adjust configuration logic accordingly. Because the status condition that the WMCO used to check for the CCM was removed, the WMCO proceeded as if a CCM was not in use. This fix removes the check. As a result, the WMCO always configures the required logic on Azure clusters. (link:https://issues.redhat.com/browse/OCPBUGS-31626[*OCPBUGS-31626*])
```

Comment:
```suggestion
Previously, on Azure clusters the WMCO would check if an external Cloud Controller Manager (CCM) was being used on the cluster. If a CCM was being used, the Operator would adjust configuration logic accordingly. Because the status condition that the WMCO used to check for the CCM was removed, the WMCO proceeded as if a CCM was not in use. This fix removes the check. As a result, the WMCO always configures the required logic on Azure clusters. (link:https://issues.redhat.com/browse/OCPBUGS-31626[*OCPBUGS-31626*])
```
- Make tenses consistent